File: programming/project-learning/python/algorithms/simple-binary-search-tree.py
# Data for the binary search tree nodes
data = ["A", "+", "B", "*", "C", "/", "D", "-", "E"]

# Left child indices for each node
left = [None, 0, None, 2, None, 1, None, 6, None]

# Right child indices for each node
right = [None, 3, None, 4, None, 7, None, 8, None]

# Root node index
root = 5

# Function to perform a post-order traversal of the binary search tree
def P(T):
    # If the left child exists, traverse the left subtree
    if left[T] is not None:
        P(left[T])
    # If the right child exists, traverse the right subtree
    if right[T] is not None:
        P(right[T])
    # Print the data of the current node
    print(data[T])

# Start the traversal from the root node
P(root)

File: programming/project-learning/php/algorithms/simple-binary-search-tree.php
<?php
// Array representing the data stored in the binary search tree nodes
$data = ["A", "+", "B", "*", "C", "/", "D", "-", "E"];

// Arrays representing the left and right children of each node
$left = [null, 0, null, 2, null, 1, null, 6, null];
$right = [null, 3, null, 4, null, 7, null, 8, null];

// Index of the root node in the $data array
$root = 5;

// Function to perform a post-order traversal of the binary search tree
function P($T, $data, $left, $right) {
    // Recursively traverse the left subtree if it exists
    if ($left[$T] !== null) {
        P($left[$T], $data, $left, $right);
    }
    // Recursively traverse the right subtree if it exists
    if ($right[$T] !== null) {
        P($right[$T], $data, $left, $right);
    }
    // Print the data of the current node
    echo $data[$T] . "\n";
}

// Start the post-order traversal from the root node
P($root, $data, $left, $right);
?>